internal/policy/store.go

```diff
@@ -4,32 +4,45 @@ import (
 	"github.com/TykTechnologies/tyk/user"
 )
 
-// Store is an in-memory policy storage object that
-// implements the repository for policy access. We
-// do not implement concurrency protections here.
+// Store is an in-memory policy storage object that implements the
+// repository for policy access. We  do not implement concurrency
+// protections here. Where order is important, use this.
 type Store struct {
```

Comment:
why is this changed?

Reply:
because custom policies doesn't preserve order (map) and would fail existing tests with a high -c run count.
preserving order is important for methods for assertion, however a more complex assertion could be made to match methods in any order, avoiding the change in /user

Reply:
Sidenote:

- This could have been a `CustomPolicies []Policy` on SessionState, not sure why it needs the API around a map[string]any;
- I'm not sure why custom policies were stuck into metadata other than to piggyback on an accessible key/value store field. No public API changes ensure whatever is coupled to it continues to work.